src/OpenRiaServices.Tools/Framework/RiaClientFilesTaskHelpers.cs

```diff
@@ -75,6 +76,47 @@ public static bool CodeGenForNet6(string generatedFileName, ClientCodeGeneration
     }
 #endif
 
+    /// <summary>
+    /// Create and write arguments to a new file
+    /// </summary>
+    /// <param name="arguments">The arguments that should be written to a file</param>
+    /// <returns>The name of the file created</returns>
+    internal static string CreateAndWriteArgumentsToNewTempFile(IEnumerable<string> arguments)
+    {
+        string fileName = Path.Combine(Path.GetTempPath(), $"openria-codegen-{DateTime.Now:yyyyMMdd-HHmmss-fff}.tmp");
+
+        try
+        {
+            WriteContentsToNewFile(fileName, arguments);
+        }
+        catch (IOException)
+        {
+            // The file could already exist, which would throw an IOExcpetion
+            // Add random file name to the file name and try again
+            fileName += Path.GetRandomFileName();
+            WriteContentsToNewFile(fileName, arguments);
```

Comment:
nit: the random part will become part of the extension, maybe a "Replace(..tmp", random + tmp)"

Reply:
I had an extra look at  the documentation and it will get another extension, so no issue